integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/datamap/FGDataMapTestCase.scala

```diff
@@ -481,6 +479,62 @@ class FGDataMapTestCase extends QueryTest with BeforeAndAfterAll {
       sql("select * from normal_test where name='n502670' and city='c2670'"))
   }
 
+  test("test fg datamap and validate the visible and invisible status of datamap ") {
```

Comment:
I think you can add a new testcaseSuite  to test the visibility of datamap and it will eliminate some duplicate code.

Besides, I've noticed that you pointed that the invisibility for lucence datamap may cause failure.  Better to add that test in the same testcaseSuite.

Reply:
lucene datamap don't support invisibility feature, please refer:https://issues.apache.org/jira/browse/CARBONDATA-2502

Reply:
Jacky said there are many test case suite, it shouldn't add more test case suite if there exists related test case suite